proto/sedachain/batching/v1/batching.proto

```diff
@@ -44,19 +49,45 @@ message Signature {
   uint32 scheme = 1;
   // signature is the signature of the batch.
   string signature = 2;
-  // public_key is the public key corresponding to the signature that 
+  // public_key is the public key corresponding to the signature that
   // should have been registered in x/pubkey
   string public_key = 3;
   // merkle_proof is the proof of inclusion of the signer in the validator
-  // tree. 
+  // tree.
   string merkle_proof = 4;
 }
 
-// Module parameters which can be changed through governance.
+// Params is a list of parameters which can be changed through governance.
 message Params {
   option (gogoproto.equal) = true;
 
   // validator_set_trim_percent is the percentage of the validator
   // set to store in the validator merkle tree in the batch.
   uint32 validator_set_trim_percent = 1;
 }
+
+// DataResult represents the result of a resolved data request.
+message DataResult {
+  // id is the Keccack-256 hash of the data result.
+  string id = 1;
+  // dr_id is the data request identifier.
+  string dr_id = 2;
+  // version is a semantic version string.
+  string version = 3;
+  // block_height is the height at which the data request was tallied.
+  uint64 block_height = 4;
+  // exit_code is the exit code of the tally wasm binary execution.
+  uint32 exit_code = 5;
+  // gas_used is the gas used by the data request execution.
+  uint64 gas_used = 6;
+  // result is the result of the tally wasm binary execution.
+  bytes result = 7;
```

Comment:
I think it was cosmwasm that converted the byte slices to base64, even though we prefer to avoid base64 representations. Since the tally vm returns byte slices, the module would have to encode them to base64 if we wanted that.

Reply:
You could directly pass byte slices to the contract iirc, we did it to make the messages/shorter and cap the size. I'm not sure if the size is a concern anymore, tbh. Might be a group question/decision.

Reply:
I think for our collective sanity it’s probably for the best if at protocol we treat bytes as slices wherever possible and as hex if we need it as a string. 
if some framework/library converts the bytes to base64 when it goes over the wire/to disk that’s fine, but we should take care to deserialise it at the boundaries.

My 2 cents 😬

Reply:
Ah I should have specified that I was talking exclusively about bytes instead of a base64 string. With regards to fixed size vs dynamic I fully agree with you :)

Reply:
I realized I might have misunderstood and you might not have been referring to the salt, whoopsies my bad. I do still get scared by passing unspecified length byte arrays though.